executor/src/main/java/org/k8loud/executor/action/Action.java

```diff
@@ -1,7 +1,13 @@
 package org.k8loud.executor.action;
 
 
-public interface Action {
+import lombok.Data;
 
-    void perform();
+import java.util.Map;
+
+@Data
+public abstract class Action {
+    private final Map<String, String> params;
+
+    public abstract void perform();
```

Comment:
It would either be empty (unnecessary layer of abstraction) or have methods signatures defined (which can be done as it is at the moment).
Imo there's no reason to add the interface.

Reply:
Ok for now let it be. In the future we will see if adding interfaces add some readibility